### PR TITLE
SI-2159-filter-AD-by-beneficiary

### PR DESCRIPTION
**Pull request by Allyson-English**
[ticket](https://linear.app/dimo/issue/SI-2159/filter-aftermarket-devices-by-beneficiary)

- Allow clients to filter the aftermarket devices list by beneficiary

### CODE REVIEW

File: internal/repositories/aftermarket_devices_test.go [hunk 1]
```diff
@@ -134,3 +137,60 @@ func Test_GetOwnedAftermarketDevices_Pagination_PreviousPage(t *testing.T) {
...
+	e := msgpack.NewEncoder(&b)
+	e.UseArrayEncodedStructs(true)
+	_ = e.Encode(aftermarketDevicePrimaryKey{TokenID: tknID})
+	id := "AD_" + base64.StdEncoding.EncodeToString(b.Bytes())
+	mn, err := gmn.EntropyToMnemonicThreeWords(aftermarketDeviceNodeMintedArgs.AftermarketDeviceAddress.Bytes())
+	if err != nil {
+		assert.NoError(t, err)
+	}
+	name := strings.Join(mn, " ")
+	expectedOwnerFilterResp := []*model.AftermarketDeviceEdge{
```
Comment (elffjs) [inline internal/repositories/aftermarket_devices_test.go:172]: No way could this pass, yeah? You're going to get all 4.

Comment (elffjs) [reply internal/repositories/aftermarket_devices_test.go:172]: I'm stupid.

**Comment by elffjs** [comment]
@0xdev22 Can you take a look?

File: internal/repositories/aftermarket_devices_test.go [hunk 2]
```diff
@@ -134,3 +134,54 @@ func Test_GetOwnedAftermarketDevices_Pagination_PreviousPage(t *testing.T) {
 	}
 	assert.Exactly(t, expected, res.Edges)
 }
+
+func Test_GetAftermarketDevices_FilterByBeneficiary(t *testing.T) {
```
Comment (0xdev22) [inline internal/repositories/aftermarket_devices_test.go:138]: Some missing tests

- When beneficiary has no records (**another beneficary's id is supplied that has no records**)
- Both Owner and Beneficiary are supplied

**Review by 0xdev22 — CHANGES_REQUESTED** [review]

File: internal/repositories/aftermarket_devices_test.go (hunk 2 above)
Comment (elffjs) [reply internal/repositories/aftermarket_devices_test.go:138]: I'm okay to go without these given how orthogonal the code is with everything else. Let's focus on tests for the other PR, which is a bit more intricate.

**Pull request merged**